File: Java/프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.ArrayDeque;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int progressCount = progresses.length;
        int[] completedDays = new int[progressCount];
        ArrayDeque<Integer> queue = new ArrayDeque<>();
        
        for (int i = 0; i < progressCount; i++) {
            int progress = progresses[i];
            int speed = speeds[i];
            int day = 0;
            while (progress < 100) {
                progress += speed;
                day++;
            }
            completedDays[i] = day;
        }
        
        int maxDay = completedDays[0];
        int feature = 0;
        for (int i = 0; i < completedDays.length; i++) {
            if (maxDay >= completedDays[i]) {
                feature++;
            } else {
                queue.addLast(feature);
                maxDay = completedDays[i];
                feature = 1;
            }
        }
        queue.addLast(feature);
        
        return queue.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: Swift/프로그래머스/2/42586. 기능개발/기능개발.swift
import Foundation

struct Queue<T> {
    private var queue: [T] = []
    
    var getQueue: [T] {
        return queue
    }
    
    var count: Int {
        return queue.count
    }
    
    mutating func add(_ element: T) {
        queue.append(element)
    }
    
    mutating func poll() -> T? {
        return queue.isEmpty ? nil : queue.removeFirst()
    }
}

func solution(_ progresses:[Int], _ speeds:[Int]) -> [Int] {
    let n = progresses.count
    var solution: Queue<Int> = Queue()
    
    var completed: [Int] = []
    for progressIndex in 0..<n {
        let progress = Double(progresses[progressIndex])
        let speed = Double(speeds[progressIndex])
        let spendedTime = Int(ceil((100.0-progress)/speed))
        completed.append(spendedTime)
    }
    
    var maxDay = completed[0]
    var feat = 0
    for day in completed {
        if day <= maxDay {
            feat += 1
        } else {
            solution.add(feat)
            maxDay = day
            feat = 1
        }
    }
    
    solution.add(feat)
    return solution.getQueue
}